src/Shared/CertificateGeneration/PemCertificateFile.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Certificates.Generation
+{
+    internal class PemCertificateFile : IDisposable
+    {
+        public string FilePath { get; }
+        public X509Certificate2 Certificate { get; }
+
+        public PemCertificateFile(X509Certificate2 certificate)
+        {
+            Certificate = certificate;
+            string directory = Paths.XdgRuntimeDir ?? Paths.Home ?? Path.GetTempPath();
+            FilePath = Path.Combine(directory, Guid.NewGuid() + ".pem");
+            string pem = new string(PemEncoding.Write("CERTIFICATE", certificate.Export(X509ContentType.Cert)));
+            File.WriteAllText(FilePath, pem);
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                File.Delete(FilePath);
+            }
+            finally
+            { }
+        }
+    }
+}
```

Comment:
I don't think we need this. There is already functionality to export a certitificate in PEM format [here](https://github.com/dotnet/aspnetcore/blob/1fba8accda5b7c4e9a3be38abfeafa14266a5cf0/src/Shared/CertificateGeneration/CertificateManager.cs#L448). Just use that instead.